### ISSUE TITLE
A few issues I’d like addressed before using this project daily. It's a great project, but these small issues need fixing.

### ISSUE THREAD

**Issue by rishabhkrmahato** [XengShi/materialYouNewTab#174]
1.
![image](https://github.com/user-attachments/assets/53d5bb06-0e87-4c9d-9215-e628cb119c9c)
There’s a space above the greetings (like "Good morning") where text can’t be entered properly. After typing two characters, the cursor jumps back to the start.

2.
![image](https://github.com/user-attachments/assets/466ed5da-6ad8-40a9-be20-964ab858037e)
Unable to add a shortcut with a URL such as: (it makes a shortcut, that leads to your repo instead)
`data:text/html, <body contenteditable style="font: 2rem/1.5 monospace;max-width:60rem;margin:0 auto;padding:4rem;[">](data:text/html, <body contenteditable style=)`

3. 
Unable to add additional GitHub links as shortcuts. 
For instance, attempting to add a shortcut to my github with a single GitHub link doesn’t work— weirdly it makes 3 shortcuts, that all default to your repo link.
![image](https://github.com/user-attachments/assets/f92624f8-8db1-419b-9e6e-c120738644f4)
![image](https://github.com/user-attachments/assets/0677cf56-9e33-4f25-8c53-ff80d70446ed)

4. 
Please add location suggestions in the input bar so users can select directly. Integrating Google Maps API could help. Also, consider replacing the “Input Options” button with a “Find Location” button.

5.
When entering a weather location in latitude and longitude format, the weather displays correctly, but the location button only shows “Earth.”
![image](https://github.com/user-attachments/assets/51c53d8c-df99-4bfc-856d-f3d969e9b6e8)
![image](https://github.com/user-attachments/assets/9f7ec4ab-6b59-4e2d-be82-53d159e8b8b9)

**Comment by itz-rj-here** [comment]
Can you also describe which version of the material You NewTab you are using, which machine you are using this extention. also, the browser name too.

**Comment by prem-k-r** [comment]
Can you please check with latest updates. You can either cloning or downloading zip file. you can download the ZIP file by clicking on the Code button. On the repository page, look for a green button labeled Code. Click on the Code button, and in the dropdown menu, select Download ZIP.

**Comment by itz-rj-here** [comment]
![image](https://github.com/user-attachments/assets/40312139-cd06-46fd-8c02-a1bfd7712bcf)
I don't see any problem on the latest code actual.

**Comment by prem-k-r** [comment]
@itz-rj-here someone else too earlier mentioned 1st issue. Was it in Mac? Which issue # was that do you know?

**Comment by itz-rj-here** [comment]
> @itz-rj-here someone else too earlier mentioned 1st issue. Was it in Mac? Which issue # was that do you know?

Let me look.

**Comment by rishabhkrmahato** [comment]
@itz-rj-here, I'm using Firefox and installed the extension from the Mozilla Add-ons page. Isn't that the latest version?

**Comment by XengShi** [comment]
> @itz-rj-here, I'm using Firefox and installed the extension from the Mozilla Add-ons page. Isn't that the latest version?

No, download it from the GitHub to get the Latest version of extension.

**Comment by itz-rj-here** [comment]
> @itz-rj-here, I'm using Firefox and installed the extension from the Mozilla Add-ons page. Isn't that the latest version?

@rishabhkrmahato No actually. As @XengShi already included about the latest version, when there is significant updates XengShi will update the extention on the store.
![image](https://github.com/user-attachments/assets/cf1bc27d-3fc6-4975-aa3a-4afe9815d770)

**Comment by itz-rj-here** [comment]
> @itz-rj-here, I'm using Firefox and installed the extension from the Mozilla Add-ons page. Isn't that the latest version?

And on the main code you can find some more changes on the extention.

**Comment by rishabhkrmahato** [comment]
@itz-rj-here , oh! my bad then, didn't see it was mentioned, will check out the github-latest one now!

**Comment by itz-rj-here** [comment]
Can you kindly download the main code files and look for the issues if they're fixed, as you mentioned previously? @rishabhkrmahato

**Comment by rishabhkrmahato** [comment]
@itz-rj-here 
Yes, I just checked, and none of those issues are present in the main GitHub version. However, as mentioned in the installation section of the README, this is a temporary add-on. So, I have to reload this temporary add-on every time I open Firefox, which makes it too cumbersome to use regularly. Could you guide me on how to make it permanent? Thanks!

**Comment by rishabhkrmahato** [comment]
but this null-html (blank html notebook that i keep on bookmark toolbar)
thing is still there issue no. 2: it still makes a shortcut and points it to this repo!
![image](https://github.com/user-attachments/assets/7c45a21b-9b30-48ce-92ac-415eac236ff8)

**Comment by itz-rj-here** [comment]
> @itz-rj-here Yes, I just checked, and none of those issues are present in the main GitHub version. However, as mentioned in the installation section of the README, this is a temporary add-on. So, I have to reload this temporary add-on every time I open Firefox, which makes it too cumbersome to use regularly. Could you guide me on how to make it permanent? Thanks!

Actually I'm a Chrome User. As far I know about Firefox there is no option to make it permanent until new version released. If you still has any questions you can freely talk to @XengShi the main Author of the extention. Cause he has a better brain than me. 😄

**Comment by itz-rj-here** [comment]
> but this null-html (blank html notebook that i keep on bookmark toolbar) thing is still there issue no. 2: it still makes a shortcut and points it to this repo! ![image](https://private-user-images.githubusercontent.com/65235557/384998186-7c45a21b-9b30-48ce-92ac-415eac236ff8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNDM0NjMsIm5iZiI6MTczMTM0MzE2MywicGF0aCI6Ii82NTIzNTU1Ny8zODQ5OTgxODYtN2M0NWEyMWItOWIzMC00OGNlLTkyYWMtNDE1ZWFjMjM2ZmY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTExVDE2MzkyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkMjY1ZGFiZmU4ZWVlOTlmMzc2ZTUwMzc2NzFiZjlhMGMwMGE4MWZlYjEyNDk3MWVlYTQ4NDhkOTNlNGRiY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l31Oh28RkjG7YvNtvpBv47EXQAQe5rEkLFeufdis4C0)

Sorry but I don't have that much idea about null-html. I think @prem-k-r can handle about this issue.

**Comment by rishabhkrmahato** [comment]
ok! thanks! Keeping this issue open for the owner/developers to address. I would love to use this daily as my Firefox new tab.

**Comment by prem-k-r** [comment]
> @itz-rj-here Yes, I just checked, and none of those issues are present in the main GitHub version. However, as mentioned in the installation section of the README, this is a temporary add-on. So, I have to reload this temporary add-on every time I open Firefox, which makes it too cumbersome to use regularly. Could you guide me on how to make it permanent? Thanks!

It required signing.. it is a complex process. 
Simpler process is available for Firefox Developer Edition and Firefox Nightly only.
I suggest waitfor  a day or two.. XengShi may release v2.4 and put it in store

**Comment by prem-k-r** [comment]
> > but this null-html (blank html notebook that i keep on bookmark toolbar) thing is still there issue no. 2: it still makes a shortcut and points it to this repo! ![image](https://private-user-images.githubusercontent.com/65235557/384998186-7c45a21b-9b30-48ce-92ac-415eac236ff8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNDM0NjMsIm5iZiI6MTczMTM0MzE2MywicGF0aCI6Ii82NTIzNTU1Ny8zODQ5OTgxODYtN2M0NWEyMWItOWIzMC00OGNlLTkyYWMtNDE1ZWFjMjM2ZmY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTExVDE2MzkyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkMjY1ZGFiZmU4ZWVlOTlmMzc2ZTUwMzc2NzFiZjlhMGMwMGE4MWZlYjEyNDk3MWVlYTQ4NDhkOTNlNGRiY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l31Oh28RkjG7YvNtvpBv47EXQAQe5rEkLFeufdis4C0)
> 
> Sorry but I don't have that much idea about null-html. I think @prem-k-r can handle about this issue.

Is it non-deletable?
editing name and link to create desired shortcut not working?

**Comment by rishabhkrmahato** [comment]
> Is it non-deletable? editing name and link to create desired shortcut not working?

Yes it is, I tried deleting it and... re-adding it, but the same thing keeps happening. I just wanted to create a blank HTML notebook shortcut. I think what's happening is that, since it's a blank HTML file, the shortcut is defaulting to the main repository link, which appears by default when creating a new shortcut."

**Comment by itz-rj-here** [comment]
@prem-k-r I think maybe raphaelritz made some mistakes on the shortcut code.

**Comment by prem-k-r** [comment]
> > Is it non-deletable? editing name and link to create desired shortcut not working?
> 
> Yes it is, I tried deleting it and... re-adding it, but the same thing keeps happening. I just wanted to create a blank HTML notebook shortcut. I think what's happening is that, since it's a blank HTML file, the shortcut is defaulting to the main repository link, which appears by default when creating a new shortcut.

So this is only happening when adding blank html shortcut. The link you are replacing to is `"null-html"`
And since there is auto prepend. 
It is redirecting to `https://null-html`
Correct?
